### ISSUE TITLE
Add optional Scattered appearance for Stacking component

### ISSUE THREAD

**Issue by Bod9001** [unitystation/unitystation#2446]
# Description
Currently there is no Sprite handling for it Just a number on the bottom right

## Detailed Bug Report
![image](https://user-images.githubusercontent.com/27101670/70394214-c1a3ca80-19ea-11ea-9f31-075955d9824e.png)
Example of what stacked ores look like

**Comment by chairbender** [comment]
@Bod9001 is this a duplicate of #2355 ? If so can you add your screenshots to that ticket and close this?

**Comment by chairbender** [comment]
So it looks like in the case of ores, it actually draws multiple instances of the ore sprites randomly, so I don't know if just having a custom sprite per amount would be good there. I wonder if metal stacks are all different sprites or are actually done with custom rendering logic, creating multiple instances of the same sprite.

Perhaps in addition to allowing different sprites per stackable amount, we could also have the ability to support simply drawing additional instances of the sprite based on amount, so we could support ores in that way. Then for something like metal you could even just say that it stacks in an ordered manner (to a max of 3) and for ores you can say it stacks in a random manner.

I'll take a look at how tg does it.

**Comment by chairbender** [comment]
stack.dm
```
/obj/item/stack/update_icon()
	if(novariants)
		return ..()
	if(amount <= (max_amount * (1/3)))
		icon_state = initial(icon_state)
	else if (amount <= (max_amount * (2/3)))
		icon_state = "[initial(icon_state)]_2"
	else
		icon_state = "[initial(icon_state)]_3"
	..()
```

So that looks like it's just a different sprite for each kind, which we should definitely support. I'll look into what's done for minerals

**Comment by Bod9001** [comment]
so  have this for minerals?

**Comment by chairbender** [comment]
no, I'm looking into what's done for ores on the ground. There might be special logic for it.

**Comment by Bod9001** [comment]
I mean this issue for minerals*

**Comment by chairbender** [comment]
Nah it's fine to do separately. This isn't really a critical part of the mineral system as once you have an ore it's basically just like any other stackable item.

**Comment by chairbender** [comment]
yup, custom logic in ores_coins.dm:
```
/obj/item/stack/ore/update_icon()
	var/difference = min(ORESTACK_OVERLAYS_MAX, amount) - (LAZYLEN(stack_overlays)+1)
	if(difference == 0)
		return
	else if(difference < 0 && LAZYLEN(stack_overlays))			//amount < stack_overlays, remove excess.
		cut_overlays()
		if (LAZYLEN(stack_overlays)-difference <= 0)
			stack_overlays = null;
		else
			stack_overlays.len += difference
	else if(difference > 0)			//amount > stack_overlays, add some.
		cut_overlays()
		for(var/i in 1 to difference)
			var/mutable_appearance/newore = mutable_appearance(icon, icon_state)
			newore.pixel_x = rand(-8,8)
			newore.pixel_y = rand(-8,8)
			LAZYADD(stack_overlays, newore)
	if (stack_overlays)
		add_overlay(stack_overlays)
```

so we could either build this into stackable itself (a flag, like "scattered appearance") or develop it as a custom component for ores. My suggestion would be for this to be a capability of stackable since other objects might want it but I'm fine with it as a custom component for ores.

**Comment by Bod9001** [comment]
I think scattered appearance would be best
allows devs to be lazy with Sprite work : P

**Comment by MaxIsJoe** [comment]
Does anyone know a good way to make a scattered appearance without creating lots of SpriteHandlers or using a particle system?
